doc/developer/platform/ux.md

```diff
@@ -1,20 +1,726 @@
+field. Naively packing the UUIDs is unwieldy, however:
+
+```
+893677c2-a74c-4ecb-804d-3b7fa0c61fdc:68104da2-3417-442e-a614-f078dac470be
+```
+
+But since UUIDs are just 16 bytes each, we can pack them into a much shorter
+base64-encoded string:
+
+```py
+>>> base64.urlsafe_b64encode(client_id.bytes + secret_key.bytes).decode().rstrip('=')
+iTZ3wqdMTsuATTt_oMYf3GgQTaI0F0QuphTweNrEcL4
+```
+
+When encoded in this form, we call the value an app-specific password rather
+than an API key.
+
+In the web UI, we will need to build a custom interface for managing personal
+API tokens that shows the app-specific password form alongside the client ID
+and secret.
+
+### Billing
+
+Materialize Platform will bill monthly based on usage. Accounts will be billed
+in terms of compute units and storage units. A strawman proposal: compute units
+are machine-hours and storage units are gigabyte-months.
+
+We expect metering the compute usage of clusters and the storage usage of
+sources and tables to be straightforward. What is not clear is how to bill for
+the overhead of other system components.
+
+### Region
+
+When an account is created, it does not have any regions. The user's first step
+after registering for a new account is to choose one region to activate.
+
+Additional regions can be enabled via a "+ Enable new region" button in the
+region selector or somesuch in the web UI, or via the [global API](#global-api).
+
+To prevent disaster, regions cannot be deactivated. Users who wish to deactivate
+a region can contact support.
+
+### Cluster
+
+Clusters are created within a region via a new `CREATE CLUSTER` SQL statement:
+
+```sql
+CREATE CLUSTER <name> SIZE {xs|s|m|l|xl} [, REPLICATION FACTOR <n>];
+```
+
+Creating a cluster allocates persistent compute resources (e.g., EC2 instances)
+and begins billing the account for those resources. The SQL statement returns
+successfully as soon as the request is received.
+
+Specifying a replication factor greater than one will run multiple identical
+replicas of the cluster for redundancy. If one replica fails, requests can
+be routed to the other replica(s) until the failed replica is brought back
+online and brought up to date.
+
+A cluster may be horizontally scaled across multiple machines. The scaling is
+deliberately hidden from users behind the opaque notion of "size".
+
+The list of clusters and the status of each can be observed via the new `SHOW
+CLUSTERS` statement or the new `mz_clusters` system catalog table. The exact
+output of these commands is TODO, but they will include the provisioning status
+of the EC2 instances.
+
+It is possible to create indexes in a cluster before it is provisioned, but
+attempts to `SELECT` from these indexes will fail until the cluster is
+provisioned.
+
+Clusters are dropped via a new `DROP CLUSTER` SQL statement:
+
+```sql
+DROP CLUSTER <name> [CASCADE];
+```
+
+Clusters cannot be dropped while they contain any indexes unless `CASCADE`
+is specified.
```

Comment:
This might be less important than in normal SQL, because you cannot have dependencies on indexes that span clusters. So, maybe similar to being able to drop a whole DATABASE at once. Not sure if you need to CASCADE that or not, but each DROP CLUSTER could implicitly be CASCADE.

Reply:
You do need to `CASCADE` your database drops in Materialize (long design discussion about this somewhere), so I think it's most consistent to have the same restriction on clusters.